unit-tests/src/test/java/org/eclipse/collections/impl/list/mutable/primitive/AbstractBooleanListTestCase.java

```diff
@@ -396,4 +403,91 @@ public void toImmutable()
         ImmutableBooleanList immutable = this.classUnderTest().toImmutable();
         Assert.assertEquals(BooleanArrayList.newListWith(true, false, true), immutable);
     }
+
+    @Test
+    public void collectByte()
+    {
+        MutableBooleanList list = this.newWith(true, false, true);
+
+        MutableByteList mutableByteList =
+                list.collectByte(each -> each ? (byte) 1 : (byte) 0);
```

Comment:
Seems this line can be simplified with
`list.collectByte(each -> each ? (byte) (each ? 1 : 0));`

And the same case for `short` and `char` below.